examples/producer_channel_example/producer_channel_example.go

```diff
@@ -58,7 +59,15 @@ func main() {
 						*m.TopicPartition.Topic, m.TopicPartition.Partition, m.TopicPartition.Offset)
 				}
 				return
+			case *kafka.Error:
+				err := ev
+				if err.IsRetriable() {
```

Comment:
Delivery errors are passed as `*kafka.Message`, see the error checking on line 54.

`*kafka.Error` is for generic client-level errors, and there is no notion of retriable there (since it is not directly tied to an API call).

Reply:
Ok, thanks for the explanation. When and how should `err.IsRetriable()` be used? Can it be ignored or not be called at all because it's not tied to an API call? Then why not remove it? I am confused.

Reply:
IsRetriable() is currently only returned by the transactional producer APIs, but the plan is to have all APIs return this eventually, where it makes sense.